### ISSUE TITLE
NA/NaN/Inf in foreign function call (arg 1)

### ISSUE THREAD

**Issue by zbc123a** [cszang/bootRes#1]
I am calculating the correlation between chronology and climate data (SPI). However,  two of the datasets come out an error:
```
Error in qr.default(a, tol = tol) : 
  NA/NaN/Inf in foreign function call (arg 1)
Called from: qr.default(a, tol = tol)
```
I have checked the input dataset. They are all numeric without na/nan/inf. I am using the latest version of BootRes.

**Comment by cszang** [comment]
Do you have a lot of exact zeros in the climate data, or very little variability, e.g. SPI is always very similar for a given month? This can lead to numerical problems with the QR-decomposition.

BTW: bootRes is not actively developed anymore. Please use [treeclim](https://github.com/cszang/treeclim) (also on CRAN).

**Comment by zbc123a** [comment]
> Do you have a lot of exact zeros in the climate data, or very little variability, e.g. SPI is always very similar for a given month? This can lead to numerical problems with the QR-decomposition.
> 
> BTW: bootRes is not actively developed anymore. Please use [treeclim](https://github.com/cszang/treeclim) (also on CRAN).

Thank you very much for your help! I checked the dataset, there is not zero value and I think the data is various through the period. I attached one of the SPI datasets. I will try the treeclim package to see whether my dataset doesn't meet the requirements.
[ca687.txt](https://github.com/cszang/bootRes/files/3282324/ca687.txt)

**Comment by cszang** [comment]
Is this the actual data file you are using? It only has seven years of data, with a high number of variables (i.e. months from current and previous year) this could get tricky...

**Comment by zbc123a** [comment]
Yes. But this period is the only overlap between the public tree-ring dataset and GRACE-DA dataset (my study target). The growing season is set up from March to October. I am wondering whether the results of this analysis are convincing.
